### PR TITLE
feat: Add a SendableContext protocol.

### PR DESCRIPTION
**Pull request by silasary**
## What type of pull request is this?

- [x] Non-breaking code change
- [ ] Breaking code change
- [ ] Documentation change/addition
- [x] Tests change

## Description
Adds a protocol that is essentially "Context + SendMixin"
This is useful for type-hinting anything that works in either Message or Interaction contexts.

## Checklist

- [x] I've formatted my code with [Black](https://black.readthedocs.io/en/stable/)
- [x] I've ensured my code works on `Python 3.10.x`
- [x] I've tested my code

### CODE REVIEW

**Comment by silasary** [comment]
Let me know if this feels like the wrong file to put it in

**Comment by LordOfPolls** [comment]
No qualms with this outside of the pre-commit fails

File: .github/workflows/pytest-pr.yml [hunk 1]
```diff
@@ -55,7 +55,7 @@ jobs:
       - name: Install pytest
         run: |
           pip install -e ${{ matrix.extras }}
-          pip install pytest pytest-recording pytest-asyncio pytest-cov
```
Comment (Kigstn) [inline .github/workflows/pytest-pr.yml:58]: Depends on how strict polls is, but technically this does not belong in this PR

Comment (LordOfPolls) [reply .github/workflows/pytest-pr.yml:58]: Not feeling strict at all

Comment (silasary) [reply .github/workflows/pytest-pr.yml:58]: I included it because I didn't want to add typeguard to both files ;)

**Review by LordOfPolls — APPROVED** [review]
